### DISCUSSION TITLE
[New Feature] Migrate to Next.js

### DISCUSSION THREAD

**Discussion by doprz** [monkeytypegame/monkeytype#2866] [Ideas] [answered]
I'm proposing migrating monkeytype to Next.js.

This comes with multiple benefits:
* Hybrid SSG and SSR
* TypeScript support
* File-system Routing
* API Routes
* Fast Refresh
* Image Optimization

@Miodec Can I be assigned this feature? 
It's a big feature so we'll probably have to wait until some PR are resolved. 
Thanks.

**Comment by Miodec** [comment, marked as answer]
As far as I can see this is a react framework - I don't want to convert to React, sorry. Surely not now.

**Comment by doprz** [comment]
Thanks for the reply. It's just a suggestion due to the benefits of React and NextJS. Thanks anyways.

**Reply by rizwanmustafa** [reply]
Well you can always fork the repo and do your own stuff if you want to.
